server/controllers/OauthController.js

```diff
@@ -0,0 +1,85 @@
+
+import { User } from '../database/models'; //eslint-disable-line
+import tokenService from '../utils/services/tokenService';
+
+
+/**
+ *
+ * The OauthController contains static methods that are used to handle the
+ * different routes used for social third-party authenication.
+ *
+ * The methods it contains are used in line with PassportJS module
+ *  @class OauthController
+ *  @returns {void} this is a class thus does not return anything
+ */
+class OauthController {
+  /**
+   * This method is used as the passport callback function
+   * for all the third party systems that were implemented.
+   *  The method persists or retrieves a user into or from
+   * the database.
+   * @param {object} accessToken an accessToken which
+   * could be used to work with the provider
+   * @param {string} refreshToken string that can be used
+   * to refresh the user token
+   * @param {object} profile an object containing the data of the user
+   * @param {function} done  fuunction that must be called once all
+   * processing is complete
+   * @author Oguejiofor Chidiebere
+  * @returns {void} void
+   */
+  static passportCallback(accessToken, refreshToken, profile, done) {
+    const email = profile.emails[0].value;
+    let { familyName: firstName, givenName: lastName } = profile.name;
+    if (!(firstName && lastName)) {
+      [firstName, lastName] = profile.displayName.split(' ');
+    }
+    const user = {
+      hash: profile.id,
```

Comment:
Using `profile.id` as hash/password value is not secure. It means anyone that has access to `profile.id` can actually login using the local strategy, since `email` and `profile.id` can easily be accessed.